internal/db/db.go

```diff
@@ -262,9 +262,41 @@ func (db *db) AddDocActorRelationship(
 		return client.AddDocActorRelationshipResult{}, err
 	}
 
+	err = db.publishDocUpdateEvent(ctx, docID, collection)
+	if err != nil {
+		return client.AddDocActorRelationshipResult{}, err
+	}
```

Comment:
todo: Current implementation as it stands will `publishDocUpdateEvent` even if the `AddDocActorRelation` call was a no-op, i.e. the relationship already existed and someone calls the command again and again.

IMO can likely check if `ExistedAlready` == `false` and publish the event only then, otherwise if the relationship existed already then it's a no-op, hence no need to publish the event.

Reply:
good catch. Adjusted.